smallrye-reactive-messaging-mqtt-server/src/main/java/io/smallrye/reactive/messaging/mqtt/server/MqttServerSource.java

```diff
@@ -0,0 +1,159 @@
+                        endpoint.will().getWillQos(), endpoint.will().isWillRetain());
+            }
+
+            logger.trace("[keep alive timeout = {}]", endpoint.keepAliveTimeSeconds());
+
+            endpoint.exceptionHandler(
+                    error -> logger.error("Error with client " + endpoint.clientIdentifier(), error));
+
+            endpoint.disconnectHandler(
+                    v -> logger.debug("MQTT client [{}] disconnected", endpoint.clientIdentifier()));
+
+            endpoint.pingHandler(
+                    v -> logger.trace("Ping received from client [{}]", endpoint.clientIdentifier()));
+
+            endpoint.publishHandler(message -> {
+                logger.debug("Just received message [{}] with QoS [{}] from client [{}]",
+                        message.payload(),
+                        message.qosLevel(), endpoint.clientIdentifier());
+
+                processor.onNext(new MqttMessage(message, endpoint.clientIdentifier(), () -> {
+                    if (message.qosLevel() == AT_LEAST_ONCE) {
+                        logger.trace("Send PUBACK to client [{}] for message [{}]",
+                                endpoint.clientIdentifier(),
+                                message.messageId());
+                        endpoint.publishAcknowledge(message.messageId());
+                    } else if (message.qosLevel() == EXACTLY_ONCE) {
+                        logger.trace("Send PUBREC to client [{}] for message [{}]",
+                                endpoint.clientIdentifier(),
+                                message.messageId());
+                        endpoint.publishReceived(message.messageId());
+                    }
+                    return CompletableFuture.completedFuture(null);
+                }));
+            });
+
+            endpoint.publishReleaseHandler(messageId -> {
+                logger.trace("Send PUBCOMP to client [{}] for message [{}]", endpoint.clientIdentifier(),
+                        messageId);
+                endpoint.publishComplete(messageId);
+            });
+
+            endpoint.subscribeHandler(subscribeMessage -> {
+                logger.trace("Received subscription message {} from client [{}], closing connection",
+                        subscribeMessage, endpoint.clientIdentifier());
+                endpoint.close();
+            });
+
+            // accept connection from the remote client
+            // this implementation doesn't keep track of sessions
+            endpoint.accept(false);
+        });
+
+        this.source = ReactiveStreams.fromPublisher(processor
+                .delaySubscription(mqttServer.rxListen()
+                        .doOnSuccess(ignored -> logger
+                                .info("MQTT server listening on {}:{}", options.getHost(), mqttServer.actualPort()))
+                        .doOnError(throwable -> logger.error("Failed to start MQTT server", throwable))
+                        .toFlowable()
```

Comment:
I believe the `)` is required here. 
This is the publisher used for the subscription delay.

Reply:
I'm not sure to understand what you mean: you want to apply the `compose()` on the result of the `delaySubscription()` call right?

Reply:
My bad, I missed a `))` at the end of the `doOnError` line.